### PR TITLE
Fix invalid tests for COM native server

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
see #20970 

cc @janvorli

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
cc @jkoritzinsky

File: tests/src/Interop/COM/NativeServer/StringTesting.h [hunk 1]
```diff
@@ -225,8 +225,11 @@ class StringTesting : public UnknownImpl, public IStringTesting
         /*[in]*/ LPWSTR a,
         /*[out]*/ LPWSTR b)
     {
-        ReverseInplace(::wcslen(b), b);
-        return S_OK;
+        // Not possible to test from native server
```
Comment (jkoritzinsky) [inline tests/src/Interop/COM/NativeServer/StringTesting.h]: Couldn't we also use the `UnmanagedType.VBByRefStr` marshalling behavior to correctly test this from the native server?

Comment (AaronRobinsonMSFT) [reply tests/src/Interop/COM/NativeServer/StringTesting.h]: Not really. This is testing the out attribute only. Changing the marshaling type is a different test. The point here is we can't properly test the out attribute on a default wide-string marshal.

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Innerloop Build please

**Pull request merged**